File: 백준/Silver/11866. 요세푸스 문제 0/요세푸스 문제 0.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
let input = fs.readFileSync(filePath)
    .toString()
    .trim()
// .split("\n")
// .map(x => x.trim())


function solution(args) {
    let [n, m] = args.split(' ').map(Number);
    const circle = Array.from({ length: n }, (_, i) => i + 1);
    const result = [];
    let index = 0;

    while (circle.length > 0) {
        index = (index + m - 1) % circle.length;
        result.push(circle[index]);
        circle.splice(index, 1);
    }
    console.log(`<${result.join(', ')}>`);
}


solution(input);

File: 백준/Silver/11866. 요세푸스 문제 0/요세푸스 문제 0.py
import sys
import os

# 운영 체제 확인 및 입력 파일 설정 (필요한 경우)
if os.name == 'nt':  # Windows 환경
    sys.stdin = open('input.txt', 'r')

n, m = map(int, input().split())

numbers = [i for i in range(1, n + 1)]
# 1 2 3 4 5 6 7 -> 3 6 2 7 5 1 4

answer = []
index = 0
while numbers:
    index = (index + m - 1) % len(numbers)
    answer.append(numbers[index])
    numbers.pop(index)

print(f'<{", ".join(map(str, answer))}>')
